navit/android.c

```diff
@@ -680,38 +680,10 @@ static void android_search_idle(struct android_search_priv *search_priv) {
     dbg(lvl_info, "leave");
 }
 
-static char *search_fix_spaces(const char *str) {
-    int i;
-    int len=strlen(str);
-    char c,*s,*d,*ret=g_strdup(str);
-
-    for (i = 0 ; i < len ; i++) {
-        if (ret[i] == ',' || ret[i] == '/')
-            ret[i]=' ';
-    }
-    s=ret;
-    d=ret;
-    len=0;
-    do {
-        c=*s++;
-        if (c != ' ' || len != 0) {
-            *d++=c;
-            len++;
-        }
-        while (c == ' ' && *s == ' ')
-            s++;
-        if (c == ' ' && *s == '\0') {
-            d--;
-            len--;
-        }
-    } while (c);
-
-    return ret;
-}
-
 static void start_search(struct android_search_priv *search_priv, const char *search_string) {
     dbg(lvl_debug,"enter %s", search_string);
-    char *str=search_fix_spaces(search_string);
+    char tmp_search_string = *search_string; // This is allows us to avoid warnings about discarding const qualifier
```

Comment:
Sorry guys, but this won't work.

The statement 
`char tmp_search_string = *search_string`
will copy the first char (8 bits) out of the char array search_string into tmp_search_string variable.
Later on
`search_fix_spaces(&tmp_search_string);`
passes the reference to this one byte variable as a parameter. This is neither the complete string nor is it zero terminated.


As the input in both versions is g_strdup'ed enyways, it should be safe to treat the input string always as const avoiding the need to do any bulls*** to remove a const.

But are you sure the both versions really behave the same? For me it doesn't really look so on the first glance. The second "do" loop seems to shift the content of "ret" to remove leading and trailing spaces?

I'd start with correctly commenting the header of both versions and use this work to analyse what both versions do.
Then I'd reconsider if and how to unify them.

Bonus: Who frees the duplicate returned as a result in both versions?